src/components/flat-table/flat-table-row/flat-table-row.style.js

```diff
@@ -226,29 +253,48 @@ const StyledFlatTableRow = styled.tr`
             }
           }
           /* Styling for safari. Position relative does not work on tr elements on Safari  */
-          @media not all and (min-resolution: 0.001dpcm) {
-            @supports (-webkit-appearance: none) and (stroke-color: transparent) {
+          ${isSafari(navigator) &&
+          css`
+            ${theme.focusRedesignOptOut &&
```

Comment:
You're right, and I was going to do it but because it would mean I have to do the ternary in a backwards order for the jest ignores combined with the fact that this is temporary - I don't think it's worth it and it's slightly easier to remove the legacy support if it's not in a ternary.